File: app/src/main/java/ru/application/homemedkit/databaseController/Alarm.java
package ru.application.homemedkit.databaseController;

import androidx.room.Entity;
import androidx.room.ForeignKey;
import androidx.room.Ignore;
import androidx.room.PrimaryKey;

@Entity(tableName = "alarms",
        foreignKeys = @ForeignKey(
                entity = Intake.class,
                parentColumns = "intakeId",
                childColumns = "intakeId",
                onUpdate = ForeignKey.CASCADE,
                onDelete = ForeignKey.CASCADE))
public class Alarm {
    @PrimaryKey(autoGenerate = true)
    public long alarmId;
    public long intakeId;
    public long trigger;

    public Alarm() {
    }

    @Ignore
    public Alarm(long intakeId, long trigger) {
        this.intakeId = intakeId;
        this.trigger = trigger;
    }
}

File: app/src/main/java/ru/application/homemedkit/data/dto/Alarm.kt
package ru.application.homemedkit.data.dto

import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.ForeignKey.Companion.CASCADE
import androidx.room.PrimaryKey

@Entity(
    tableName = "alarms",
    foreignKeys = [
        ForeignKey(
            entity = Intake::class,
            parentColumns = ["intakeId"],
            childColumns = ["intakeId"],
            onUpdate = CASCADE,
            onDelete = CASCADE
        )
    ]
)
data class Alarm(
    @PrimaryKey(autoGenerate = true)
    val alarmId: Long = 0L,
    val intakeId: Long = 0L,
    val trigger: Long = 0L,
    val amount: Double = 0.0,
    val preAlarm: Boolean = false
)